spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -574,14 +575,13 @@ else if (exchangeType.equals(ExchangeTypes.HEADERS)) {
 					}
 				}
 				else {
-					Assert.isTrue(value instanceof String, "value must resolve to a String for type conversion");
-					String valueString = (String) value;
-					if (StringUtils.hasText(valueString)) {
-						if (CONVERSION_SERVICE.canConvert(String.class, typeClass)) {
-							map.put(key, CONVERSION_SERVICE.convert(valueString, typeClass));
+					if (value instanceof String && StringUtils.hasText((String) value)) {
```

Comment:
We still have this `String` restriction...
All other types are ignored for the `CS`.

If you worry here about the `String value() default "";`, how about do not `resolveExpression(arg.value())` if it is an empty? Therefore `value == null` and we are fine with the `map.put()`.
OTOH it is end-user problem that his SpEL evaluates to empty String, not `null`.